app/models/deploy_service.rb

```diff
@@ -5,8 +5,8 @@ def initialize(user)
     @user = user
   end
 
-  def deploy!(stage, reference)
-    deploy = stage.create_deploy(reference: reference, user: user)
+  def deploy!(stage, attributes = {})
+    deploy = stage.create_deploy(user: user, **attributes)
```

Comment:
maybe make the interface be `user, attributes` ... then we don't need to merge or slice